src/App.js

```diff
@@ -1,24 +1,52 @@
-import logo from './logo.svg';
-import './App.css';
+import React from "react";
+import "./App.css";
+import Product from "./components/Product/Product";
+import Search from "./components/Search/Search";
+import Loader from "./utils/loader/loader";
+// import products from "./data/product.js"
 
 function App() {
+  const [products, setProducts] = React.useState(null);
+
+  const [search, setSearch] = React.useState("");
+  const [error, setError] = React.useState(null);
+
+  React.useEffect(() => {
+    fetch("https://dummyjson.com/products")
+      .then((response) => response.json())
+      .then((data) => {
+        setProducts(data.products);
+      })
+      .catch((error) => {
+        setError("Error while fetching data");
+      });
+  }, []);
+
+  const product = () => {
+    // error
+    if (error) {
+      return <div className="error-card">{error}</div>;
+    }
+    // loading
+    if (products === null) {
```

Comment:
Sir I was actually confused, which one to use ( null or [] (empty array) ). 
because they all represents different things.
null      -> haven't received anything (may be URL incorrect)..
[]         -> URL correct but after fetching no product is received ( 0 product )..
[ {...} ] -> URL correct and have products.

Can you please tell me if "null" is better or "[]" to initialise ?